### PR TITLE
ci: track json schema versions

### PR DESCRIPTION
**Pull request by jjenscodee**
## Description

- #62 

This PR adds the "$id" property of the schema to define the base URI with a version identifier and introduce a new workflow to:
1. looks for changes to the version in the $id of the top-level schema
2. bundles the schemas into a single flat
3. push the file to the `docs/static/schemas` under a path that matches the retrieval URI of the top-level $id

## Review guidelines

Estimated Time of Review: 8 minutes

## Checklist before merging:

- [x] Confirm that the relevant changelog(s) are up-to-date in case of any user-facing changes

### CODE REVIEW

File: .github/actions/bundle-schema/action.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      if: ${{ steps.versions.outputs.new_version != steps.versions.outputs.prev_version }}
+      shell: bash
+      run: |
+        python3 utils/bundle.py ${{ inputs.schema-path }} ${{ inputs.schema }}
+        mkdir -p docs/static/schemas/${{ steps.versions.outputs.version_only }}
+        mv ${{ inputs.schema }} docs/static/schemas${{ steps.versions.outputs.new_version }}
+        git config --global user.name github-actions[bot]
+        git config --global user.email github-actions[bot]@users.noreply.github.com
+        git add .
+        git commit -m "generate new schema"
```
Comment (eeberhard) [inline .github/actions/bundle-schema/action.yml:52]:
```suggestion
        git commit -m "docs: ${{ inputs.schema }} ${{ steps.versions.outputs.version_only }}"
```

File: schemas/component-descriptions/schema/component.schema.json
Comment (eeberhard) [inline schemas/component-descriptions/schema/component.schema.json:1]: could you also delete the old id tags from the subschemas in this directory? the `predicate`, `service` and `signal_collection` sub-schemas don't need their own id afterwards.

File: .github/workflows/bundle-schema.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      - name: Bundle & Push JSON schema for application schemas
+        uses: ./.github/actions/bundle-schema
+        with:
+          schema-path: ./schemas/applications/schema/application.schema.json
+          schema: application.schema.json
+      - name: Bundle & Push JSON schema for component-descriptions schemas
+        uses: ./.github/actions/bundle-schema
+        with:
+          schema-path: ./schemas/component-descriptions/schema/component.schema.json
+          schema: component.schema.json
```
Comment (eeberhard) [inline .github/workflows/bundle-schema.yml:26]: Could you also update the `$id` of the controller description schema and include it here?

```suggestion
          schema: component.schema.json
      - name: Bundle & Push JSON schema for controller-descriptions schemas
        uses: ./.github/actions/bundle-schema
        with:
          schema-path: ./schemas/controller-descriptions/schema/controller.schema.json
          schema: controller.schema.json
```

**Comment by jjenscodee** [comment]
> Just to check, the bundle-schema action will bundle and push commits directly to main right? 

Yes, that's what I intend to do

File: .github/actions/bundle-schema/action.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      shell: bash
+      run: |
+        python3 utils/bundle.py ${{ inputs.schema-path }} ${{ inputs.schema }}
+        mkdir -p docs/static/schemas/${{ steps.versions.outputs.version_only }}
+        mv ${{ inputs.schema }} docs/static/schemas${{ steps.versions.outputs.new_version }}
+        git config --global user.name github-actions[bot]
+        git config --global user.email github-actions[bot]@users.noreply.github.com
+        git add .
+        git commit -m "docs: ${{ inputs.schema }} ${{ steps.versions.outputs.version_only }}"
+        git push
```
Comment (LouisBrunner) [inline .github/actions/bundle-schema/action.yml:53]: It pushes on the same branch?

Comment (jjenscodee) [reply .github/actions/bundle-schema/action.yml:53]: yes it pushes on the same branch. There's a condition for the workflow:
`
on:
  push:
    branches:
      - main
`
so I think the workflow happens on the main branch and this push will push the changes to main, maybe I'm wrong? @LouisBrunner

Comment (LouisBrunner) [reply .github/actions/bundle-schema/action.yml:53]: Yes, that's correct! I missed the `on` condition, thanks.

I don't like when the CI pushes on branches used by humans as it is a pain to use (have to pull everytime you push, etc) but it's a good v1 and we are happy with the downsides for now I think.

**Review by LouisBrunner — APPROVED** [review]

File: .github/actions/bundle-schema/action.yml (hunk 3 above)
Comment (eeberhard) [reply .github/actions/bundle-schema/action.yml:53]: yes I think the benefits of this approach outweigh the downsides and it's a good v1

**Pull request merged**